src/features/health/agent.controller.js

```diff
@@ -0,0 +1,106 @@
+const { Agent } = require('../../shared/db/mongodb/schemas');
+
+const createAgent = async (req, res) => {
+    const { first_name, last_name, email, region } = req.body;
+
+    //check if req fields provided
+    if (!first_name || !last_name || !email || !region) {
+        return res.status(400).send({
+            success: false,
+            message: 'first_name, last_name, email and region are required'
+        });
+    }
+
+    //Set sales to 0 and default values
+    const agent = await Agent.create ({
+        first_name,
+        last_name,
+        email,
+        region,
+        sales: 0, // makes sales default to 0
+        //any additional defaults would go here
+    });
+    if (!agent){
+        res.status(500).send({message: 'Create failed'});
+    }
+    //return agent object
+    res.status(201).send({
+        success: true,
+        message: 'Agent created successfully',
+        agent
+    });
+};
+  
+const getAgents =  async (req, res) => {
+    try {
+        const agents = await Agent.find().sort({ last_name: 1 });
+        res.json(agents);
+    } catch (error) {
+        res.status(500).send({ message: 'Error fetching agents', error });
+    }
+};
+
+const agentsByRegion = async (req, res) => {
+    const { region } = req.query;
+    if (!region) {
+        return res.status(400).send({ message: 'Region is required' });
+    }
+
+    try {
+        const agents = await Agent.find({ region: region }).sort({ rating: -1 });
+        res.json(agents);
+    } catch (error) {
+        res.status(500).send({ message: 'Error fetching agents', error });
+    }
+};
+
+const updateAgents = async (req, res) => {
+    const {first_name, last_name, email, region} = req.body;
+    if (!email) {
+        return res.status(400).send({message: 'Email is required'});
+    }
+
+    try{
+        const agent = await Agent.findOneAndUpdate(
+            {email: email},
```

Comment:
Not an error, but just curious. Why did you decide to sort by email instead of by its id? This looks fine as is but the implication on this is that you could never update the email of the agent which might be problematic in a live-action setting.

Reply:
I was thinking that the email was the only unique way to identify the agents since first or last names could be the same. I completely forgot that once they were added to the database they were assigned id numbers! Thank you, I'll change things around here.